src/store/ui/useStudioUIStore.ts

```diff
@@ -26,43 +20,51 @@ export const useStudioUIStore = create<StudioUIStore>()(
     recordings: new Map(),
     generators: new Map(),
     scripts: new Map(),
+    data: new Map(),
     proxyStatus: 'offline',
 
-    addFile: (path) =>
+    addFile: (file) =>
       set((state) => {
-        const file = fileFromFileName(path)
-
         if (file.type === 'recording') {
-          state.recordings.set(path, file)
+          state.recordings.set(file.fileName, file)
         }
 
         if (file.type === 'generator') {
-          state.generators.set(path, file)
+          state.generators.set(file.fileName, file)
         }
 
         if (file.type === 'script') {
-          state.scripts.set(path, file)
+          state.scripts.set(file.fileName, file)
+        }
+
+        if (file.type === 'data') {
+          state.data.set(file.fileName, file)
         }
       }),
-    removeFile: (path) =>
+    removeFile: (file) =>
       set((state) => {
```

Comment:
It now seems to me that it'd better to store all files in a flat map and get `recordings`, `generators, `scripts`, etc. using a zustand selector